src/view/com/post-thread/PostThread.tsx

```diff
@@ -163,42 +169,75 @@ function PostThreadLoaded({
     [threadViewPrefs, thread],
   )
 
-  // construct content
+  // On web we always are ready to render, we only need this on mobile.
+  const [readyToShowAll, setReadyToShowAll] = React.useState(isWeb)
+  const [topPageCount, setTopPageCount] = React.useState(1)
+  const isPrepending = React.useRef(false)
+
   const posts = React.useMemo(() => {
-    let arr = Array.from(
-      flattenThreadSkeleton(
-        sortThread(thread, threadViewPrefs),
-        hasSession,
-        treeView,
-      ),
+    const items = createThreadSkeleton(
+      sortThread(thread, threadViewPrefs),
+      hasSession,
+      treeView,
+      readyToShowAll,
     )
+
+    const highlightedPost = isThreadPost(items.highlightedPost)
+      ? items.highlightedPost
+      : undefined
+
+    // Create the posts to prepend
+    const postsToPrepend = [
+      ...(!highlightedPost?.ctx.isParentLoading ? [TOP_COMPONENT] : []),
+      ...(!isWeb
+        ? [...items.parents.slice(-(topPageCount * 15))]
+        : [...items.parents]),
+    ]
+
```

Comment:
Posts to prepend has these possibilities:

1. We have the highlighted post, and we see the highlighted post has no parents. Therefore, we add the TOP_COMPONENT here.
2. We have the highlighted post, and we see that there is more posts that will be added above, so we don't add the TOP_COMPONENT and let the generator add it instead.
3. If we are on web, we just render every child. There is no reason to simulate the load on web.
4. On native, we add 15 posts per page. Each time the `onStartReached` callback is fired, `topPageCount` will be updated if necessary.